lnwallet/chancloser/rbf_coop_transitions.go

```diff
@@ -867,38 +1270,214 @@ func (l *LocalCloseStart) ProcessEvent(event ProtocolEvent, env *Environment,
+		partialSig := wireSigWithNonce.PartialSig
+
+		if noClosee {
+			taprootPartialSigs.CloserNoClosee = tlv.SomeRecordT(
+				tlv.NewRecordT[tlv.TlvType5](partialSig),
+			)
+		} else {
+			taprootPartialSigs.CloserAndClosee = tlv.SomeRecordT(
+				tlv.NewRecordT[tlv.TlvType7](partialSig),
+			)
+		}
+
+		// Generate our next closee nonce for the next RBF iteration.
+		// This is the nonce the closer should use for our closee
+		// signature in the next RBF round. We always include this since
+		// RBF could occur.
+		nextNonces, err := env.RemoteMusigSession.ClosingNonce()
+		if err != nil {
+			return nil, fmt.Errorf("failed to generate next "+
+				"closee nonce: %w", err)
+		}
+		nextCloseeNonce = tlv.SomeRecordT(
+			tlv.NewRecordT[tlv.TlvType22](
+				lnwire.Musig2Nonce(nextNonces.PubNonce),
+			),
+		)
+	} else {
+		// Non-taproot: use regular signatures.
+		if noClosee {
+			closingSigs.CloserNoClosee = newSigTlv[tlv.TlvType1](
+				wireSig,
+			)
+		} else {
+			closingSigs.CloserAndClosee = newSigTlv[tlv.TlvType3](
+				wireSig,
+			)
+		}
+	}
+
+	return &lnwire.ClosingSig{
+		ChannelID:          env.ChanID,
+		CloserScript:       remoteScript,
+		CloseeScript:       localScript,
+		FeeSatoshis:        fee,
+		LockTime:           lockTime,
+		ClosingSigs:        closingSigs,
+		TaprootPartialSigs: taprootPartialSigs,
+		NextCloseeNonce:    nextCloseeNonce,
+	}, nil
+}
+
+// extractTaprootPartialSig extracts just the PartialSig from TaprootPartialSigs.
+// This is useful when we need the actual partial sig for combining.
+func extractTaprootPartialSig(sigs lnwire.TaprootPartialSigs) (
+	partialSig fn.Option[lnwire.PartialSig]) {
+
+	if sigs.CloserNoClosee.IsSome() {
+		var ps lnwire.PartialSig
+		sigs.CloserNoClosee.WhenSomeV(func(p lnwire.PartialSig) {
+			ps = p
+		})
+		return fn.Some(ps)
+	}
+
+	if sigs.NoCloserClosee.IsSome() {
+		var ps lnwire.PartialSig
+		sigs.NoCloserClosee.WhenSomeV(func(p lnwire.PartialSig) {
+			ps = p
+		})
+		return fn.Some(ps)
+	}
+
+	if sigs.CloserAndClosee.IsSome() {
+		var ps lnwire.PartialSig
+		sigs.CloserAndClosee.WhenSomeV(func(p lnwire.PartialSig) {
+			ps = p
+		})
+		return fn.Some(ps)
+	}
+
+	return fn.None[lnwire.PartialSig]()
+}
+
+// prepareClosingSignatures prepares the local and remote signatures for the
+// closing transaction. For taproot channels, it handles musig signature
+// combination. For non-taproot channels, it converts wire signatures to regular
+// signatures.
+func prepareClosingSignatures(env *Environment, l *LocalOfferSent,
+	msg *LocalSigReceived, sig lnwire.Sig,
+	closeOpts []lnwallet.ChanCloseOpt,
+) (localSig, remoteSig input.Signature, err error) {
+
+	if env.IsTaproot() {
+		// For taproot channels, we need to reconstruct the
+		// MusigPartialSig from the wire signature. We'll need to create
+		// a new CreateCloseProposal to get the proper MusigPartialSig
+		// that CompleteCooperativeClose expects.
+		rawLocalSig, _, _, err := env.CloseSigner.CreateCloseProposal(
```

Comment:
Q: is `CreateCloseProposal` idempotent? Thinking about that we may call this twice, once in `LocalCloseStart` to send the offer, then here when combining the sigs. I guess if the tests are passing this should work.